extension/otto/gohan_util.go

```diff
@@ -84,6 +85,63 @@ func init() {
 				value, _ := vm.ToValue(resp)
 				return value
 			},
+			"gohan_raw_http": func(call otto.FunctionCall) otto.Value {
+				VerifyCallArguments(&call, "gohan_raw_http", 4)
+				method, err := GetString(call.Argument(0))
+				if err != nil {
+					ThrowOttoException(&call, err.Error())
+				}
+				url, err := GetString(call.Argument(1))
+				if err != nil {
+					ThrowOttoException(&call, err.Error())
+				}
+				rawHeaders, err := GetMap(call.Argument(2))
+				if err != nil {
+					ThrowOttoException(&call, err.Error())
+				}
+				rawData, err := GetString(call.Argument(3))
+				if err != nil {
+					ThrowOttoException(&call, err.Error())
+				}
+				//TODO: pass Transport options like timeouts
+
+				// prepare request
```

Comment:
The rest of the function below could be a separate `GohanRawHTTP()` function, like [pull #218@Refactor builtins:GohanDbQuery](https://github.com/cloudwan/gohan/pull/218/commits/9780d7929938862e144f66b078018787cb3f8a77#diff-ff4e9f745acd7902b3d865025f2589bbR284).